zebra-network/src/peer_set/initialize.rs

```diff
@@ -158,10 +158,12 @@ where
     // Connect peerset_tx to the 3 peer sources:
     //
     // 1. Incoming peer connections, via a listener.
-    let listen_guard = tokio::spawn(
-        accept_inbound_connections(tcp_listener, listen_handshaker, peerset_tx.clone())
-            .instrument(Span::current()),
-    );
+    let listen_fut = {
+        let config = config.clone();
+        let peerset_tx = peerset_tx.clone();
+        accept_inbound_connections(config, tcp_listener, listen_handshaker, peerset_tx)
+    };
```

Comment:
Minor and low-priority: I'm a bit curious about style preferences :sweat_smile: How would you compare to:

```suggestion
    let listen_fut = accept_inbound_connections(
        config.clone(),
        tcp_listener,
        listen_handshaker,
        peerset_tx.clone(),
    );
```

Reply:
Looks better!

When we were instrumenting the future immediately, the extra block was required. But now it's redundant.

See 0fb5bc93e where I did this refactor for a few different function calls.